Cleanup README.md (#7947)
fabricjs/fabric.js: README.md

<a href="http://fabricjs.com">FabricJS.com</a> is a **simple and powerful Javascript HTML5 canvas library**. It is also an **SVG-to-canvas parser**.

<!-- build/coverage status, climate -->
[![Build Status](https://secure.travis-ci.org/fabricjs/fabric.js.svg?branch=master)](http://travis-ci.org/#!/kangax/fabric.js)
[![Code Climate](https://d3s6mut3hikguw.cloudfront.net/github/kangax/fabric.js/badges/gpa.svg)](https://codeclimate.com/github/kangax/fabric.js)
[![Gitpod Ready-to-Code](https://img.shields.io/badge/Gitpod-Ready--to--Code-blue?logo=gitpod)](https://gitpod.io/#https://github.com/fabricjs/fabric.js)

<!-- npm, bower, CDNJS versions, downloads -->
[![CDNJS version](https://img.shields.io/cdnjs/v/fabric.js.svg)](https://cdnjs.com/libraries/fabric.js)
[![NPM version](https://badge.fury.io/js/fabric.svg)](http://badge.fury.io/js/fabric)
[![Downloads per month](https://img.shields.io/npm/dm/fabric.svg)](https://www.npmjs.org/package/fabric)
[![Bower version](https://badge.fury.io/bo/fabric.svg)](http://badge.fury.io/bo/fabric)

<br/>

> Using Fabric.js, you can create and populate objects on canvas; objects like simple geometrical shapes — rectangles, circles, ellipses, polygons, or more complex shapes consisting of hundreds or thousands of simple paths. You can then scale, move, and rotate these objects with the mouse; modify their properties — color, transparency, z-index, etc. You can also manipulate these objects altogether — grouping them with a simple mouse selection. Fabric.js will also allow you to convert an SVG image into JavaScript data that can be used for putting it onto the `<canvas>` element.

Reviewer: I think this should be written in bullets

Reviewer: And what about adding some gifs?

<br />

<a href="http://fabricjs.com/fabric-intro-part-1">Introduction</a>
<span>&nbsp;&nbsp;•&nbsp;&nbsp;</span>

> <a href="http://fabricjs.com/docs/">Docs</a>

Reviewer: I don't think this part is needed, github parses the page and creates a navbar

Author: [https://github.com/reactchartjs/react-chartjs-2](https://github.com/reactchartjs/react-chartjs-2) was my "inspiration" here. 😄 I didn't know that Github creates navbar somewhere. Where I can find it?

Reviewer: In the README page.
![צילום מסך 2022-05-17 233113](https://user-images.githubusercontent.com/34343793/168904786-a4a69179-d0c3-4ce3-8b88-9009e625032e.png)

import { fabric } from "fabric";
```

> NOTE: If you are using Fabric.js in a Node.js script, you will depend from [node-canvas](https://github.com/Automattic/node-canvas).`node-canvas` is an HTML canvas replacement that works on top of native libraries. Please [follow the instructions](https://github.com/Automattic/node-canvas#compiling) to get it up and running.

Reviewer: ```suggestion
NOTE: If you are using Fabric.js in a Node.js script, you will depend on [node-canvas](https://github.com/Automattic/node-canvas). `node-canvas` is an HTML canvas replacement that works on top of native libraries. Please [follow the instructions](https://github.com/Automattic/node-canvas#compiling) to get it up and running.
```

I wonder if the instructions are up to date.

Author: My suggestion is to remove it completely and leave the link to it somewhere.

> NOTE: es6 imports won't work in browser or with bundlers which expect es6 module like vite. Use commonjs syntax instead.

Reviewer: Hmm this will soon be wrong THANKFULLY

### Usage

#### Plain HTML

```html
<canvas id="canvas" width="300" height="300"></canvas>

<!-- Get latest version: https://cdnjs.com/libraries/fabric.js -->
<script src="lib/fabric.js"></script> 
<script>
    var canvas = new fabric.Canvas('canvas');

    var rect = new fabric.Rect({
        top : 100,
        left : 100,
        width : 60,
        height : 70,
        fill : 'red'
    });
    canvas.add(rect);
</script>
```

> #### ReactJS

Reviewer: This is great

```tsx
import React, {useEffect, useRef} from 'react'
import {fabric} from 'fabric'

const FabricJSCanvas = () => {
  const canvasEl = useRef(null)
  useEffect(() => {
    const canvas = new fabric.Canvas(canvasEl.current);
    const rect = new fabric.Rect({
            top : 100,
            left : 100,
            width : 60,
            height : 70,
            fill : 'red'
    });
    
    canvas.add(rect);
    return () => {
      canvas.dispose()
    }
  }, []);
  
  return (<canvas width="300" height="300" ref={canvasEl}/>)
});

export default FabricJSCanvas;
```

NOTE: Fabric.js requires a `window` object.

> <h3 id="fabric-building">Building</h3>

Reviewer: I vote to get rid of this entire section for 3 main reasons:
1. too esoteric
2. there's a new cli for fabric that is very simple to use - PRs should be made to make it better instead of this bloated part about building options.
3. The old build script is at it's end of life and will soon die out.

Author: I also vote for removing this part or move it to another document and just leave the link to it somewhere.

    1.2 You can also include all modules like so:

          $ node build.js modules=ALL

    1.3 You can exclude a few modules like so:

          $ node build.js modules=ALL exclude=gestures,image_filters

2. Create a minified distribution file

        # Using YUICompressor (default option)
        $ node build.js modules=... minifier=yui

        # or Google Closure Compiler
        $ node build.js modules=... minifier=closure

3. Enable AMD support via require.js (requires uglify)

        $ node build.js requirejs modules=...

4. Create source map file for better productive debugging (requires uglify or google closure compiler).<br>More information about [source maps](http://www.html5rocks.com/en/tutorials/developertools/sourcemaps/).

        $ node build.js sourcemap modules=...

    If you use google closure compiler you have to add `sourceMappingURL` manually at the end of the minified file all.min.js (see issue https://code.google.com/p/closure-compiler/issues/detail?id=941).

        //# sourceMappingURL=fabric.min.js.map

5. Ensure code guidelines are met (prerequisite: `npm -g install eslint`)

        $ npm run lint && npm run lint_tests

> <h3 id="fabric-building">Testing</h3>

Reviewer: Should this (and all the rest) be:
```suggestion
# Testing
```

Author: This is also a good idea. I left all those HTML tags with unchanged IDs because I wasn't sure if somebody didn't link to this paragraph somewhere else (fabricjs.com?)

Reviewer: I don't think there are...
Bad practice IMO.
I think it's safe to remove

> #### 1. Install NPM packages

Reviewer: Well the testing instructions are bad.
I think it's better to refer to CONTRIBUTING.md

```
$ npm install
```

> #### 2. Install testem

Reviewer: not required, it's installed in devs

- [Presentation from BK.js](http://www.slideshare.net/kangax/fabricjs-building-acanvaslibrarybk)
- [Presentation from Falsy Values](http://www.slideshare.net/kangax/fabric-falsy-values-8067834)

## Credits

- [@kangax](https://twitter.com/kangax)
- [Andrea Bogazzi](https://twitter.com/AndreaBogazzi) for help with bugs, new features, documentation, GitHub issues
- Ernest Delgado for the original idea of [manipulating images on canvas](http://www.ernestdelgado.com/archive/canvas/)
- [Maxim "hakunin" Chernyak](http://twitter.com/hakunin) for ideas, and help with various parts of the library throughout its life
- [Sergey Nisnevich](http://nisnya.com) for help with geometry logic
- [Stefan Kienzle](https://twitter.com/kienzle_s) for help with bugs, features, documentation, GitHub issues
- [Shutterstock](http://www.shutterstock.com/jobs) for the time and resources invested in using and improving Fabric.js
- [And all the other GitHub contributors](https://github.com/kangax/fabric.js/graphs/contributors)

## Sponsor authors

- https://flattr.com/@kangax
- https://github.com/sponsors/asturur
- https://www.patreon.com/fabricJS

## History

> Fabric.js started as a foundation for design editor on [printio.ru](http://printio.ru) — interactive online store with ability to create your own designs. The idea was to create [Javascript-based editor](http://printio.ru/ringer_man_tees/new), which would make it easy to manipulate vector shapes and images on T-Shirts. Since performance was one of the most critical requirements, we chose canvas over SVG. While SVG is excellent with static shapes, it's not as performant as canvas when it comes to dynamic manipulation of objects (movement, scaling, rotation, etc.). Fabric.js was heavily inspired by [Ernest Delgado's canvas experiment](http://www.ernestdelgado.com/public-tests/canvasphoto/demo/canvas.html). In fact, code from Ernest's experiment was the foundation of an entire framework. Later, Fabric.js grew into a collection of distinct object types and got an SVG-to-canvas parser.

Reviewer 2: ```suggestion
Fabric.js started as a foundation for design editor on [printio.ru](http://printio.ru) — interactive online store with ability to create your own designs. The idea was to create [Javascript-based editor](http://printio.ru/ringer_man_tees/new), which would make it easy to manipulate vector shapes and images on T-Shirts. Since performance was one of the most critical requirements, we chose canvas over SVG. While SVG is excellent with static shapes, it's not as performant as canvas when it comes to dynamic manipulation of objects (movement, scaling, rotation, etc.). Fabric.js was heavily inspired by a canvas experiment created by Ernest Delgado. In fact, code from Ernest's experiment was the foundation of an entire framework. Later, Fabric.js grew into a collection of distinct object types and got an SVG-to-canvas parser.
```

This canvas experiment URL is actually long dead and I can find no new location for it so probably best to just remove the link.

Reviewer 2: actually nevermind, @asturur is right this whole section is kind of unnecessary

> Yes History can be removed, we should just credit that demo developer in the credits.

    node build.js modules=ALL exclude=json no-strict no-svg-export

> ## Goals

Reviewer: I think this is more `Guidelines` than `Goals` because they are acheived.

Author: Achievements? 😄  Section contains great stuff, but I don't know how to name it properly.

- Follows [Semantic Versioning](http://semver.org/)

### Supported browsers

- Firefox 4+
- Safari 5+
- Opera 9.64+
- Chrome (all versions)
- Edge (chromium based, all versions)
- IE11 and Edge legacy, not supported. Fabric up to 5.0 is written with ES5 in mind, but no specific tests are run for those browsers.

You can [run automated unit tests](http://fabricjs.com/test/unit/) right in the browser.

### More resources

- [Fabric.js on Twitter](https://twitter.com/fabricjs)
- [Fabric.js on Google Groups](https://groups.google.com/group/fabricjs)
- [Fabric.js on CodeTriage](https://www.codetriage.com/kangax/fabric.js)
- [Fabric.js on Bountysource](https://www.bountysource.com/trackers/23217-fabric-js)
- [Fabric.js on Stackoverflow](https://stackoverflow.com/questions/tagged/fabricjs)
- [Fabric.js on jsfiddle](https://jsfiddle.net/user/fabricjs/fiddles/)
- [Fabric.js on Codepen.io](https://codepen.io/tag/fabricjs)
- [Presentation from BK.js](http://www.slideshare.net/kangax/fabricjs-building-acanvaslibrarybk)

> - [Presentation from Falsy Values](http://www.slideshare.net/kangax/fabric-falsy-values-8067834)

Reviewer: These 2 last links are really old...

Reviewer 3: Bountysource too i deleted the account after they changed the policies and eat up all the bounties

Author: Done

> ### Credits

Reviewer: This is old and outdated as well
I think it's irrelevant in the README - github does it why better.
